borg/hashindex.pyx

```diff
@@ -175,9 +192,39 @@ cdef class ChunkIndex(IndexBase):
 
     def __contains__(self, key):
         assert len(key) == self.key_size
-        data = <int *>hashindex_get(self.index, <char *>key)
+        data = <int32_t *>hashindex_get(self.index, <char *>key)
         return data != NULL
 
+    def incref(self, key):
+        """Increase refcount for 'key', return (refcount, size, csize)"""
+        assert len(key) == self.key_size
+        data = <int32_t *>hashindex_get(self.index, <char *>key)
+        if not data:
+            raise KeyError(key)
+        cdef int32_t refcount = _le32toh(data[0])
+        if refcount != INT32_MAX:
+            if INT32_MAX - refcount < 1:
```

Comment:
The check is in fact correct, it is needed for corrupted repos from before this change.
So INT32_MAX - (int32_t)-2 (or any other negative number, really) is some negative number.
It's either this or we need to make sure that refcount is > 0 (I don't think there's such a check on load?)